service/s3/internal/customizations/s3100continue_test.go

```diff
@@ -0,0 +1,153 @@
+package customizations_test
+
+import (
+	"context"
+	"crypto/tls"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/internal/awstesting"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/google/go-cmp/cmp"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// unit test for service/internal/s3shared/s3100continue.go
+func TestAdd100ContinueHttpHeader(t *testing.T) {
```

Comment:
**fix**: Is there not a way to have the test live closer to where the code that defines it is (e.g. `s3shared`)? It would also seem like we don't really need to spinup a test server for this and can instead directly test the middleware ([example](https://github.com/aws/smithy-go/blob/main/transport/http/middleware_content_length_test.go) from smithy-go).

Reply:
If I put it under `s3shared` containing middleware, circular dependencies occur since `s3shared` and `s3` depends on each other in that case due to the test.

Reply:
Right but if you don't make the test depend on the s3 client and instead just test the middleware in isolation then it should be fine (see the example tests linked in previous comment)